tests/bundle/Integration/ObjectStateTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\IbexaScheduledVisibility\Tests\Integration;
+
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\Handler\ObjectState;
+
+final class ObjectStateTest extends BaseTest
+{
+    /**
+     * @dataProvider provideCases
+     */
+    public function testUpdateVisibility(array $configuration, bool $expectedHidden)
+    {
+        $scheduledVisibilityService = $this->getScheduledVisibilityService();
+        $content = $this->createContent($configuration['publish_from'], $configuration['publish_to']);
+        $hiddenObjectState = $this->createObjectState();
+        $visibleObjectStateId = 1;
+        $objectStateGroupId = 2;
+        $handler = $this->getObjectStateHandler($objectStateGroupId, $hiddenObjectState->id, $visibleObjectStateId);
+        if ($scheduledVisibilityService->shouldBeHidden($content) && !$handler->isHidden($content)) {
+            $handler->hide($content);
+        }
+        if ($scheduledVisibilityService->shouldBeVisible($content) && $handler->isHidden($content)) {
+            $handler->reveal($content);
+        }
+        $content = $this->getRepository()->getContentService()->loadContent($content->contentInfo->getId());
+
+        $objectStateService = $this->getRepository()->getObjectStateService();
+        $objectStateGroup = $objectStateService->loadObjectStateGroup(2);
+        $objectState = $this->getRepository()->sudo(
+            static fn (): \Ibexa\Contracts\Core\Repository\Values\ObjectState\ObjectState => $objectStateService->getContentState($content->contentInfo, $objectStateGroup),
```

Comment:
Import this.

Reply:
fixed in 192a844247df8f976fd5b07880f317d09d13da2b